lib/lambda_deployment/configuration.rb

```diff
@@ -8,13 +8,14 @@ def load_config(config_file)
       config = YAML.load_file(config_file)
       @project = config.fetch('project')
       @region = config.fetch('region', ENV.fetch('AWS_REGION', nil))
-      @file_path = File.expand_path(config.fetch('file_name'), File.dirname(config_file))
-      raise "File not found: #{@file_path}" unless File.exist?(@file_path)
 
-      @s3_bucket = config.fetch('s3_bucket', ENV.fetch('LAMBDA_S3_BUCKET', nil))
       @s3_key = s3_key_name(config.fetch('file_name'))
```

Comment:
Don't you want the object in the S3 bucket to have a tag or git sha in the name?

Otherwise, if you say in Samson to deploy and specify a git tag of `v123` or a `production` branch, how do you know it's deploying that version? Or how do you roll back to a previous version?

Reply:
Yes, you and @grosser raise a very good point. In addition to checking if the file is there we should also check that the proper version is set as well. The `head_object` will return that information. We just need a way to set it in the first place, but I guess that is just a matter of generating a VERSION file, add it to the GCR build and fill the TAG env variable with the same value.

Reply:
If the name of the bucket is something like `zendesk_lambdas`, then I would name the object something like `${project_name}/lambda-${git_sha}.zip`.  That way all the objects exist in the same "directory" in the bucket, and they each have a unique name with the git sha in it.

Reply:
why not `${project_name}/${git_sha}/lambda.zip`? I have the feeling that this would be easier to deal in the path rather than the filename

Reply:
Sure, that works too.

In S3, I'm pretty sure the only 2 pieces of information are the bucket name and the name of the key. Using forward-slashes in a key are useful when using the S3 UI, but S3 doesn't really have a distinction between "directory" and "filename".